source/Body.h

```diff
@@ -99,48 +122,58 @@ class Body {
 	Point position;
 	Point velocity;
 	Angle angle;
-	Point center;
 	Point rotatedCenter;
 	// A zoom of 1 means the sprite should be drawn at half size. For objects
 	// whose sprites should be full size, use zoom = 2.
 	float zoom = 1.f;
-	float scale = 1.f;
+	mutable float scale = 1.f;
+	mutable bool debug = false;
 
 	double alpha = 1.;
 
 	// Government, for use in collision checks.
 	const Government *government = nullptr;
 
+	// Conditions registered from bodies.
+	ConditionsStore conditions;
 
 private:
-	// Set what animation step we're on. This affects future calls to GetMask()
-	// and GetFrame().
-	void SetStep(int step) const;
+	// Set what animation step we're on and indicate whether a state transition has finished.
+	// This affects future calls to GetMask() and GetFrame().
+	bool SetStep(int step) const;
 
 
 private:
 	// Animation parameters.
-	const Sprite *sprite = nullptr;
+	mutable SpriteParameters sprites[Body::BodyState::NUM_STATES] = {SpriteParameters(), SpriteParameters(),
+																SpriteParameters(), SpriteParameters(),
+																SpriteParameters(), SpriteParameters()};
+	mutable Body::BodyState currentState = Body::BodyState::FLYING,
+							transitionState = Body::BodyState::FLYING;
```

Comment:
```suggestion
	mutable BodyState currentState = BodyState::FLYING;
	mutable BodyState transitionState = BodyState::FLYING;
```
(see one of my previous comments)